roxie/ccd/ccdfile.cpp

```diff
@@ -291,12 +293,15 @@ class CRoxieLazyFileIO : implements ILazyFileIO, implements IDelayedFile, public
                             current.setown(f->open(IFOread));
                         if (current)
                         {
+                            open = true;
                             if (doTrace(traceRoxieFiles))
                                 DBGLOG("Opening %s", sourceName);
                             if (useRemoteResources)
                                 disconnectRemoteIoOnExit(current);
                             break;
                         }
+                        else
+                            setFailure();
```

Comment:
Are we holing crit here ?

Reply:
Yes, _checkOpen is used internally by member functions that have already obtained the crit lock